plugins/engines/EngineFactory.cpp

```diff
@@ -33,70 +33,46 @@
 #include <plugins/engines/livre/LivreEngine.h>
 #endif
 
+namespace
+{
+template <typename T, typename... Args>
+std::unique_ptr<T> make_unique(Args&&... args)
+{
+    return std::unique_ptr<T>(new T(std::forward<Args>(args)...));
+}
+}
+
 namespace brayns
 {
-EngineFactory::EngineFactory(int argc, const char** argv,
+EngineFactory::EngineFactory(const int argc, const char** argv,
                              ParametersManager& parametersManager)
-    : _parametersManager(parametersManager)
+    : _argc{argc}
+    , _argv{argv}
+    , _parametersManager{parametersManager}
 {
-    for (int i = 0; i < argc; ++i)
-        _arguments.push_back(argv[i]);
 }
 
-EnginePtr EngineFactory::get(const std::string& name)
+std::unique_ptr<Engine> EngineFactory::create(const std::string& name)
 {
-    if (_engines.find(name) != _engines.end())
-        return _engines[name];
     try
     {
 #if (BRAYNS_USE_OSPRAY)
         if (name == "ospray")
-        {
-            const char** argv = new const char*[_arguments.size()];
-            for (size_t i = 0; i < _arguments.size(); ++i)
-                argv[i] = _arguments[i].c_str();
-            _engines[name] = EnginePtr(
-                new OSPRayEngine(_arguments.size(), argv, _parametersManager));
-            delete[] argv;
-            return _engines[name];
-        }
+            return make_unique<OSPRayEngine>(_argc, _argv, _parametersManager);
```

Comment:
iirc, the reason for the copy of argv is that engines might mutate it, aka if the engine is not taking a const char*. Please check that, or we have a look together.

Reply:
I think that only the LivreEngine had a constness problem, but it is already making a copy of the args internally... I seriously hope that no engine actually alters the CL parameters though!